internal/generator/fluentd/output/http/http.go

```diff
@@ -82,6 +84,13 @@ func Headers(o logging.OutputSpec) Element {
 	return KV("headers", utils.ToHeaderStr(o.Http.Headers, "%q:%q"))
 }
 
+func ContentType(o logging.OutputSpec) string {
+	if o.Http == nil || len(o.Http.Headers) == 0 || o.Http.Headers["Content-Type"] == "" {
+		return "application/x-ndjson"
+	}
+	return o.Http.Headers["Content-Type"]
```

Comment:
@alanconway we can not allow setting an arbitrary content type because users only control the receiver and have literally no control over how the payload is crafted, unless @vparfonov can identify fluent is able to handle them based on this value.  As an example, a user can set any of the following:
```
 application/java-archive
 application/EDI-X12   
 application/EDIFACT   
 application/javascript   
 application/octet-stream   
 application/ogg   
 application/pdf  
 application/xhtml+xml   
 application/x-shockwave-flash    
 application/json  
 application/ld+json  
 application/xml   
 application/zip  
 application/x-www-form-urlencoded 
 ```
 I'm pretty certain fluent doesn't speak "pdf" or "javascript".   Our config crafts and manipulates JSON.  The only reasonable options IMO are: JSON, NJSON.  Does fluent convert JSON to say CSV or tab delimited based upon content-type?  If it does not we should consider adding an enum.

Reply:
@jcantrill  Make sense, but i think checking will be done on plugin side according to this table https://docs.fluentd.org/output/http#content_type

Reply:
Oh, I see, you mean we will support only 2 options `application/json` and `application/x-ndjson`

Reply:
we do not set the formatter which dictates the content type.  The default is JSON.  I think it is reasonable that in the future we may allow other formats but we should:
* default one (e.g. json)
* allow an enum on the CR of: json,ndjson

I would check first to see if vector supports both json, ndjson before we make a final decision

Reply:
@jcantrill 
As I understand it, Vector defines Content-Type independently:
https://github.com/vectordotdev/vector/blob/master/src/sinks/http.rs#L339C29-L376

In same Vector allow overriding content-type header: https://github.com/vectordotdev/vector/pull/15625 

Vector also did something specific for Datadog output: https://github.com/vectordotdev/vector/blob/master/src/sinks/datadog/logs/sink.rs#L106

Reply:
@jcantrill I take your point, but I think it makes sense to  use the content-type and reject any content type that we don't recognize or that is not appropriate in the context. If we add new API enum to indicate a content type, then we're likely going to end up having to extend it for one special case, then another, until it becomes a mini-content-type of its own.

Reply:
If we wish to accept 'content-type' then we should add a validation to support what you suggest.  It should only allow the 2 which are applicable

Reply:
agree, i will update